localstack/services/firehose/provider.py

```diff
@@ -0,0 +1,610 @@
+    def create_delivery_stream(
+        self,
+        context: RequestContext,
+        delivery_stream_name: DeliveryStreamName,
+        delivery_stream_type: DeliveryStreamType = DeliveryStreamType.DirectPut,
+        kinesis_stream_source_configuration: KinesisStreamSourceConfiguration = None,
+        delivery_stream_encryption_configuration_input: DeliveryStreamEncryptionConfigurationInput = None,
+        s3_destination_configuration: S3DestinationConfiguration = None,
+        extended_s3_destination_configuration: ExtendedS3DestinationConfiguration = None,
+        redshift_destination_configuration: RedshiftDestinationConfiguration = None,
+        elasticsearch_destination_configuration: ElasticsearchDestinationConfiguration = None,
+        amazonopensearchservice_destination_configuration: AmazonopensearchserviceDestinationConfiguration = None,
+        splunk_destination_configuration: SplunkDestinationConfiguration = None,
+        http_endpoint_destination_configuration: HttpEndpointDestinationConfiguration = None,
+        tags: TagDeliveryStreamInputTagList = None,
+    ) -> CreateDeliveryStreamOutput:
+        region = FirehoseBackend.get()
+
+        destinations: DestinationDescriptionList = []
+        if elasticsearch_destination_configuration:
+            destinations.append(
+                DestinationDescription(
+                    DestinationId=short_uid(),
+                    ElasticsearchDestinationDescription=convert_es_config_to_desc(
+                        elasticsearch_destination_configuration
+                    ),
+                )
+            )
+        if amazonopensearchservice_destination_configuration:
+            destinations.append(
+                DestinationDescription(
+                    DestinationId=short_uid(),
+                    AmazonopensearchserviceDestinationDescription=convert_opensearch_config_to_desc(
+                        amazonopensearchservice_destination_configuration
+                    ),
+                )
+            )
+        if s3_destination_configuration or extended_s3_destination_configuration:
+            destinations.append(
+                DestinationDescription(
+                    DestinationId=short_uid(),
+                    S3DestinationDescription=convert_s3_config_to_desc(
+                        s3_destination_configuration
+                    ),
+                    ExtendedS3DestinationDescription=convert_extended_s3_config_to_desc(
+                        extended_s3_destination_configuration
+                    ),
+                )
+            )
+        if http_endpoint_destination_configuration:
+            destinations.append(
+                DestinationDescription(
+                    DestinationId=short_uid(),
+                    HttpEndpointDestinationDescription=convert_http_config_to_desc(
+                        http_endpoint_destination_configuration
+                    ),
+                )
+            )
+
```

Comment:
It looks like not all of the possible destinations are accounted for here in this `destinations.append()` logic (e.g. Splunk, Redshift). If I'm following this correctly, it looks like you could invoke `create_delivery_stream()` with arguments for these non-implemented destinations, but the Firehose would make no attempt to deliver to them. Is that accurate? If so, would it be worth emitting some kind of warning in these cases?

Reply:
Correct. In LocalStack we currently only support those four destinations.
Good point to the warning, I'll one :)